media/live-seekable-range.js

```diff
@@ -1,68 +1,88 @@
-const video = document.createElement('video');
+const video = document.querySelector('video');
 
 const mediaSource = new MediaSource();
 video.src = URL.createObjectURL(mediaSource);
   
 mediaSource.addEventListener('sourceopen', function() {
   URL.revokeObjectURL(video.src);
 
-  const videoUrl = 'https://storage.googleapis.com/media-session/sample.webm';
-  const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
-
-  // Makes video behave like a live stream.
-  mediaSource.duration = +Infinity;
+  mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
 
   log('Fetching video init segment...');
-  fetch(videoUrl, { headers: { range: 'bytes=0-299' } })
-  .then(response => response.arrayBuffer())
-  .then(data => {
-
-    // Append video init segment.
-    sourceBuffer.appendBuffer(data);
-    sourceBuffer.addEventListener('updateend', function() {
-      logTimeRanges();
-
-      log('Fetching video segment that starts at 3 seconds...');
-      fetch(videoUrl, { headers: { range: 'bytes=567140-1196488' } })
-      .then(response => response.arrayBuffer())
-      .then(data => {
+  fetchAndAppendMediaSegment('bytes=0-299');
+});
 
-        // Append video segment that starts at 3 seconds.
-        sourceBuffer.appendBuffer(data);
-        sourceBuffer.addEventListener('updateend', function() {
-          logTimeRanges();
-        }, { once: true });
+function onSetInfiniteDurationButtonClick() {
+  log('User clicked "mediaSource.duration = +Infinity" button');
+  // Makes video behave like a live stream.
+  mediaSource.duration = +Infinity;
+  logMediaInfo();
+}
 
-      });
-    }, { once: true });
-  });
-});
+function onFetchAndAppend3To6MediaButtonClick() {
+  log('Fetching video segment that starts at 3 seconds...');
+  fetchAndAppendMediaSegment('bytes=567140-1196488');
+}
 
-function onSetLiveSeekableRangeButtonClick() {
-  log('User clicked "mediaSource.setLiveSeekableRange(40, 42)" button');
-  mediaSource.setLiveSeekableRange(40 /* start */, 42 /* end */);
-  logTimeRanges();
+function onSetLiveSeekableRangeButtonClick(event) {
+  const re = /mediaSource\.setLiveSeekableRange\((\d+), (\d+)\)/;
+  const start = Number(re.exec(event.target.textContent)[1]);
+  const end = Number(re.exec(event.target.textContent)[2]);
+  log(`User clicked "mediaSource.setLiveSeekableRange(${start}, ${end})" button`);
+  mediaSource.setLiveSeekableRange(start, end);
```

Comment:
I've wrapped all button clicks events in try/catch to log errors so that user understand what is happening properly. Thanks for catching this!